lib/Parse/ParseExpr.cpp

```diff
@@ -3691,10 +3691,10 @@ ParserResult<Expr> Parser::parseExprGradientBody(ExprKind kind) {
   if (consumeIf(tok::comma)) {
     // If 'withRespectTo' is used, make the user change it to 'wrt'.
     if (Tok.getText() == "withRespectTo") {
-      SourceRange withRespectToRange(Tok.getLoc(), Tok.getRange().getEnd());
+      SourceRange withRespectToRange(Tok.getLoc(), Tok.getRange().getStart());
```

Comment:
I don't quite understand this.
`Tok.getLoc()` should be equal to `Tok.getRange().getStart()`